docker/Dockerfile

```diff
@@ -0,0 +1,21 @@
+
+FROM astral/uv:bookworm-slim
+
+ARG CARROT_VERSION
+ENV CARROT_VERSION=${CARROT_VERSION}
```

Comment:
Given we want to build from source - we don't need the version number as an argument here, as versioning will be the job of `pyproject.toml`, and then the subsequent publish image workflow.